moveit_core/collision_detection/src/world.cpp

```diff
@@ -136,6 +136,100 @@ bool World::hasObject(const std::string& object_id) const
   return objects_.find(object_id) != objects_.end();
 }
 
+bool World::knowsTransform(const std::string& object_id) const
+{
+  // Check object names first
+  if (objects_.find(object_id) != objects_.end())
+    return true;
+  else  // Then objects' subframes
+  {
+    for (const auto& object : objects_)
+    {
+      // Strip the object's name that should be prepended to the requested frame
+      int prefix_length = object.second->id_.length() + 1;
+      try
+      {
+        if (object.second->subframe_poses_.find(object_id.substr(prefix_length)) != object.second->subframe_poses_.end())
+          return true;
```

Comment:
If I'm reading this correctly, this isn't actually checking if the object name is the same as in the `object_id`, just if it's the same length, i.e. If `screw` has a subframe `tip`, and `stick` doesn't have any subframes, a query for `stick/tip` will incorrectly return true.

Reply:
Haha, wow, what an awful mistake. I added explicit checks.